### PR TITLE
ROCANA-3437: Rocana configuration should support Double values

### PR DESCRIPTION
**Pull request by joey**
This is a re-submission of PR https://github.com/scalingdata/rocana-configuration/pull/12 rebased on master and updated based on the discussion there.

### CODE REVIEW

File: src/test/java/com/rocana/configuration/TestConfigurationParser.java [hunk 1]
```diff
@@ -304,4 +305,27 @@ public void testInvalidSyntax4() throws Exception {
...
+
+    try (InputStream inputStream = Resources.getResource("conf/double-requires-type.conf").openStream()) {
+      FlatObject object = parser.parse(
+        new InputStreamReader(
+          inputStream
+        ),
+        FlatObject.class
+      );
+
+      Assert.fail("Expected a ConfiguraitonException for unexpected type 'Float'");
```
Comment (basharaj) [inline src/test/java/com/rocana/configuration/TestConfigurationParser.java]: Shouldn't this say Double?

Comment (basharaj) [inline src/test/java/com/rocana/configuration/TestConfigurationParser.java]: nit: Typo in ConfiguraitonException

File: src/test/java/com/rocana/configuration/TestConfigurationParser.java [hunk 2]
```diff
@@ -304,4 +305,27 @@ public void testInvalidSyntax4() throws Exception {
...
+    } catch (ConfigurationException ex) {
+      Assert.assertTrue("Unexpected message in ConfigurationException: " + ex.getMessage(), ex.getMessage().startsWith("Unexpected type 'Float' at 23:2"));
```
Comment (basharaj) [inline src/test/java/com/rocana/configuration/TestConfigurationParser.java]: I am a little confused here, are you expecting that `float-value: 1.0` has a type or `double-value: 1.0` or both?

**Comment by basharaj** [comment]
+1 pending minor comments

File: src/test/java/com/rocana/configuration/TestConfigurationParser.java (hunk 2 above)
Comment (joey) [inline src/test/java/com/rocana/configuration/TestConfigurationParser.java]: I'm expecting the double to not work (that's what's on line 23). I have `float-value: 1.0` before `double-value: 1.0` to ensure that the constant `1.0` is of type float. That's why the expected message check looks for the line number. Should I add a comment to make that more clear?

**Pull request merged**